src/Components/DecisionCanvas/DecisionCanvasUpdated.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useMemo, useReducer } from "react";
+import { produce } from "immer";
+import Button from "@material-ui/core/Button";
+import styled from "styled-components";
+import CanvasCard from "./CanvasCard";
+
+const SectionWrapper = styled.div`
+  text-align: left;
+`;
+
+const Spacer = styled.div`
+  width: 16px;
+`;
+
+const CardBody = styled.div`
+  margin-bottom: 16px;
+  .answers {
+    font-weight: normal;
+    h3 {
+      font-size: 14px;
+    }
+  }
+  .questions {
+    color: #888888;
+    ol {
+      list-style-position: inside;
+      padding: 0;
+    }
+    h4 {
+      color: #000;
+      font-size: 14px;
+      font-weight: bold;
+      margin-top: 0;
+    }
+  }
+  h4 {
+    font-weight: normal;
+  }
+`;
+
+const CanvasHeader = styled.div`
+  display: flex;
+  button {
+    height: fit-content;
+    padding: 10px 16px;
+    text-transform: none;
+  }
+  h2 {
+    margin-right: auto;
+  }
+`;
+
+function expandArrayReducer(expandedArr, { type, index }) {
+  return produce(expandedArr, (draftExpanded) => {
+    switch (type) {
+      case "TOGGLE":
+        draftExpanded[index] = !expandedArr[index];
+        break;
+      case "EXPAND":
+        draftExpanded[index] = true;
+        break;
+      case "EXPAND_ALL":
+        draftExpanded.fill(true);
+        break;
+      case "COLLAPSE":
+        draftExpanded[index] = false;
+        break;
+      case "COLLAPSE_ALL":
+        draftExpanded.fill(false);
+        break;
+      default:
+    }
+  });
+}
```

Comment:
This is really nice! Thoughts on extracting to a component by itself? Is this type of expanding logic useful in other places? (initial thought is forms, but we have a working system over there already so probably not worth to refactor)

Reply:
This wasn't a part of my changes 😅  but I'm more than happy to extract it to a component if we think it's re-usable though!

Reply:
Haha, all good. When you say it wasn't a part of your changes, does that mean it was duplicate code from elsewhere? In that case, if it's being used in multiple files, I'd vouch for it to be extracted even more. @GodGreg - thoughts?

Reply:
From what I can see, it's only being used in `DecisionCanvas.jsx` (which is the old version of this component that we'll be keeping for legacy applications) and `DecisionCanvasUpdated.jsx` (which is what I just added).

Reply:
To add to this, Brittany and I decided it would be best to migrate from multiple DBs to a single one so that would probably affect the old DecisionCanvas file in the future. I don't think we need to worry about extracting yet as it may not be needed